src/Adapter/Pgsql.php

```diff
@@ -12,4 +15,23 @@ public function setClientTimezone(Connection $db)
 
         return $this;
     }
+
+    public function registerQueryBuilderCallbacks(QueryBuilder $queryBuilder)
+    {
+        $queryBuilder->on(QueryBuilder::ON_ASSEMBLE_SELECT, function (Select $select) {
+            $groupBy = $select->getGroupBy();
+            if (! empty($groupBy)) {
+                // All SELECT columns must appear in the GROUP BY clause or be used in an aggregate function.
+                $candidates = [];
+                foreach ($select->getColumns() as $alias => $column) {
+                    if ($column instanceof Expression) {
+                        // Assume an aggregate function here.
+                        continue;
+                    }
+                    $candidates[] = is_int($alias) ? $column : $alias;
+                }
+                $select->groupBy(array_diff($candidates, $groupBy));
```

Comment:
Empty lines after closed control structures please.

Reply:
There are two control structures if I'm not mistaken.

Reply:
🐘

Reply:
Maybe now? 😆